### ISSUE TITLE
PlotAndSave() error

### ISSUE THREAD

**Issue by CameronBodine** [dbuscombe-usgs/doodle_labeller#3]
Hey Dan,
Giving this a go on some SSS imagery.  Everything was running great until the PlotAndSave() function.  Here's the console output:

```
0%|                                                                                            | 0/1 [00:00<?, ?it/s]C:\Users\csb67\.conda\envs\doodler\lib\site-packages\rasterio\__init__.py:218: NotGeoreferencedWarning: Dataset has no geotransform set. The identity matrix may be returned.
  s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
Number of bands detected in image: 3

Initial brush width = 5
  decrease using the [1] key
  increase using the [2] key
Cycle classes with [ESC] key
Go back a frame with [b] key
Skip a frame with [s] key
Undo the current annotation with [z] key
[INFO] annotations saved to data/label_images\forDoodler_sand_flat_sand_rd_gravel_wood_rocky_fine_veg_bank.npy
100%|███████████████████████████████████████████████████████████████████████████████████| 1/1 [11:31<00:00, 691.72s/it]
[INFO] Sparse labelling complete ...
[INFO] Dense labelling ... this may take a while
[INFO] Working on forDoodler
[INFO] Found 16 image chunks
[INFO] Imagery is small (<%i px in all dimensions), so not using chunks 10000
Number of bands detected in image: 3

Large class imbalance detected ... modifying  weights accordingly
100%|████████████████████████████████████████████████████████████████████████████████████| 5/5 [03:06<00:00, 37.40s/it]
Applying median filter of size: 6
Traceback (most recent call last):
  File "doodler.py", line 313, in <module>
    PlotAndSave(img.copy(), resr, l.astype(np.uint8), prob, name, config, class_str, profile)
  File "e:\NAU\Python\Doodler\doodle_labeller\funcs.py", line 934, in PlotAndSave
    cb.ax.set_yticklabels(config['classes'], fontsize=6)
  File "C:\Users\csb67\.conda\envs\doodler\lib\site-packages\matplotlib\axes\_base.py", line 63, in wrapper
    return get_method(self)(*args, **kwargs)
  File "C:\Users\csb67\.conda\envs\doodler\lib\site-packages\matplotlib\cbook\deprecation.py", line 451, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\csb67\.conda\envs\doodler\lib\site-packages\matplotlib\axis.py", line 1793, in _set_ticklabels
    return self.set_ticklabels(labels, minor=minor, **kwargs)
  File "C:\Users\csb67\.conda\envs\doodler\lib\site-packages\matplotlib\axis.py", line 1715, in set_ticklabels
    "The number of FixedLocator locations"
ValueError: The number of FixedLocator locations (7), usually from a call to set_ticks, does not match the number of ticklabels (6).
```

And the contents of my config file:
`{
  "image_folder" : "data/images",
  "label_folder": "data/label_images",
  "max_x_steps": 4,
  "max_y_steps": 4,
  "ref_im_scale": 0.8,
  "lw": 5,
  "im_order": "RGB",
  "alpha": 0.5,
  "create_gtiff": "false",
  "num_bands": 1,
  "apply_mask": "None",
  "classes": {
   "sand_flat": "#0bec0b",
   "sand_rd": "#f12063",
   "gravel": "#7c7878",
   "wood": "#c0d03b",
   "rocky_fine": "#62894d",
   "veg_bank": "#1c6d1d"
   }
}`

Let me know if you need any other info.

**Comment by CameronBodine** [comment]
Wait, I think it is because I didn't have any annotations in one of the classes that I had in the config file.  That's likely why.

**Comment by ddbuscombe** [comment]
Good bug -I'll add error checing to make sure # annotation classes == # config classes

**Comment by npucino** [comment]
I had the same error but was not related to the non-annotated classes.
Rather I fixed it by changing **line 933** of PlotAndSave function in funcs.py into:
`cb.set_ticks(0.5 + np.arange(len(config['classes'])))`

Basically, I just removed the `+1` in the np.arange function, which caused to create a list of integers from 0 to the total number of classes in the config file +1, which doesn't match with the following **line 934**:
`cb.ax.set_yticklabels(config['classes'], fontsize=6)`

This easy fix worked for me for now. In my case I had no empty classes. If in case of empty classes an error will be produced, then it is possible that a simple if statement that checks whether empty classes are present will fix this.
